src/lib.rs

```diff
@@ -194,21 +269,118 @@ mod tests {
 
         let res = invariant.send(ADMIN, InvariantAction::AddFeeTier(fee_tier));
         assert!(!res.main_failed());
-        
-        let state: InvariantState = invariant.read_state(InvariantStateQuery::GetFeeTiers).expect("Failed to read state");
+
+        let state: InvariantState = invariant
+            .read_state(InvariantStateQuery::GetFeeTiers)
+            .expect("Failed to read state");
         assert_eq!(state, InvariantState::QueriedFeeTiers(vec![fee_tier_value]));
 
         let res = invariant.send(ADMIN, InvariantAction::AddFeeTier(fee_tier));
         assert!(!res.main_failed());
-        assert!(res.contains(&(ADMIN, InvariantEvent::ActionFailed(InvariantError::FeeTierAlreadyExist).encode())));
-
-        let state: InvariantState = invariant.read_state(InvariantStateQuery::GetFeeTiers).expect("Failed to read state");
+        assert!(res.contains(&(
+            ADMIN,
+            InvariantEvent::ActionFailed(InvariantError::FeeTierAlreadyExist).encode()
+        )));
+
+        let state: InvariantState = invariant
+            .read_state(InvariantStateQuery::GetFeeTiers)
+            .expect("Failed to read state");
         assert_eq!(state, InvariantState::QueriedFeeTiers(vec![fee_tier_value]));
 
         let res = invariant.send(ADMIN, InvariantAction::RemoveFeeTier(fee_tier));
         assert!(!res.main_failed());
 
-        let state: InvariantState = invariant.read_state(InvariantStateQuery::GetFeeTiers).expect("Failed to read state");
+        let state: InvariantState = invariant
+            .read_state(InvariantStateQuery::GetFeeTiers)
+            .expect("Failed to read state");
         assert_eq!(state, InvariantState::QueriedFeeTiers(vec![]));
     }
+
+    #[test]
+    fn test_add_pool() {
```

Comment:
Creating pool is fully permissionless, so anyone can create new pool. It will be better to use a normal user (`ActorId` which is not stored as admin) for most tests. We can use `ADMIN` as signer in cases where `ADMIN` has access to a specific entrypoint